photutils/psf/groupstars.py

```diff
@@ -0,0 +1,228 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Module which provides classes to perform source grouping."""
+
+from __future__ import division
+import abc
+from astropy.extern import six
+import numpy as np
+from astropy.table import Column, Table, vstack
+
+
+__all__ = ['DAOGroup', 'DBSCANGroup', 'GroupStarsBase']
+
+
+@six.add_metaclass(abc.ABCMeta)
+class GroupStarsBase(object):
```

Comment:
Oh, and it should be very clear whether or not `__call__` needs to be implemented.  I see below that `DAOGroup` uses `__call__` instead of `group_stars`.

My suggestion would be for this class (`GroupStarsBase`) to implement just this:
```
def __call__(self, starlist):
    return self.group_stars(starlist)
```

And then have the subclasses *only* implement `group_stars`.  That way the "function"-like capability that @hamogu suggested is preserved, but subclasses have to use the `group_stars` method name, which is much more readable.